pkg/auth/providers/local/local_auth_provider.go

```diff
@@ -207,10 +205,11 @@ func (l *Provider) SearchPrincipalsDedupe(searchKey, principalType string, token
 	var localGroups []*v3.Group
 	var err error
 
+	queryKey := strings.ToLower(searchKey)
```

Comment:
Q: Should we produce non-lowercase indexes if the search term is now always lowecase?